ssl/statem/statem_srvr.c

```diff
@@ -1246,6 +1246,12 @@ MSG_PROCESS_RETURN tls_process_client_hello(SSL *s, PACKET *pkt)
     }
     /* Check if this is actually an unexpected renegotiation ClientHello */
     if (s->renegotiate == 0 && !SSL_IS_FIRST_HANDSHAKE(s)) {
+        if ((s->options & SSL_OP_NO_RENEGOTIATION)) {
+            /* would we even get here? */
```

Comment:
I think this is probably a "should not happen" scenario...we should have prevented this in the record layer.